src/python/TaskWorker/Actions/DagmanCreator.py

```diff
@@ -366,12 +366,16 @@ def __init__(self, config, crabserver, procnum=-1, rucioClient=None):
     def populateGlideinMatching(self, info):
         scram_arch = info['tm_job_arch']
         # Set defaults
-        info['desired_arch'] = "X86_64"
+        info['required_arch'] = "X86_64"
         m = re.match("([a-z]+)(\d+)_(\w+)_(\w+)", scram_arch)
         if m:
             _, _, arch, _ = m.groups()
-            if arch == "amd64":
-                info['desired_arch'] = "X86_64"
+            if arch not in SCRAM_TO_ARCH:
+                msg = "Job configured to a ScramArch: '{}' not supported in TaskWorker".format(item)
```

Comment:
can't you use simple f-format string here ?

Reply:
sure, I added that line before we decided to use fstrings everywhere, i was trying to be consistent with existing code. but you are right, we should look only towards the future, and use fstrings only :)